lib/data/model/crop_entity.dart

```diff
@@ -35,45 +37,62 @@ class CropEntity {
-    companionPlants: extractListOfString(cropDocument, "companionPlants"),
-    complexity: extractComplexity(cropDocument),
-    content: cropDocument.data["content"],
-    cropsInRotation: extractListOfString(cropDocument, "cropsInRotation"),
-    cropType: cropTypeValues.map[cropDocument.data["cropType"]],
-    imagePathReference: cropDocument.data["image"].first.path,
-    imageUrl: "",
-    name: cropDocument.data["name"],
-    nonCompanionPlants: extractListOfString(cropDocument, "nonCompanionPlants"),
-    profitability: extractProfitability(cropDocument),
-    setupCost: extractSetupCost(cropDocument),
-    soilType: extractListOfString(cropDocument, "soilType"),
-    stages: "",
-    status: statusValues.map[cropDocument.data["status"]],
-    summary: cropDocument.data["summary"],
-    waterRequirement: extractWaterRequirements(cropDocument),
-  );
+  factory CropEntity.cropFromDocument(DocumentSnapshot cropDocument) =>
+      CropEntity(
+        companionPlants: extractListOfString(cropDocument, "companionPlants"),
+        complexity: extractComplexity(cropDocument),
+        content: (cropDocument.data["content"] != null) ? cropDocument.data["content"] : "",
+        cropsInRotation: extractListOfString(cropDocument, "cropsInRotation"),
+        cropType: cropTypeValues.map[cropDocument.data["cropType"]],
+        imagePathReference: cropDocument.data["image"].first.path,
+        imageUrl: "",
+        name: cropDocument.data["name"],
+        nonCompanionPlants:
+            extractListOfString(cropDocument, "nonCompanionPlants"),
+        profitability: extractProfitability(cropDocument),
+        setupCost: extractSetupCost(cropDocument),
+        soilType: extractListOfString(cropDocument, "soilType"),
+        stages: List(),
+        stagesPathReference: extractStagesPaths(cropDocument),
+        status: statusValues.map[cropDocument.data["status"]],
+        summary: cropDocument.data["summary"],
+        waterRequirement: extractWaterRequirements(cropDocument),
+      );
 
   /*
    ## Image url is referenced from Firebase Storage, thus, needs to be deducted from firestore's Document, asynchronously and then saved.
    */
   void setImageUrl(String imageUrl) {
     this.imageUrl = imageUrl;
   }
+
+  void addStage(Stage stage) {
+    this.stages.add(stage);
+  }
 }
 
-List<String> extractListOfString(DocumentSnapshot document, String valueToBeExtracted){
-  if(document.data[valueToBeExtracted] != null) {
+List<String> extractListOfString(
+    DocumentSnapshot document, String valueToBeExtracted) {
+  if (document.data[valueToBeExtracted] != null && document.data[valueToBeExtracted] != "") {
     return List<String>.from(document.data[valueToBeExtracted].map((x) => x));
   }
   return null;
 }
 
+List<String> extractStagesPaths(DocumentSnapshot document) {
+  if (document.data["stages"] != null) {
+    return List<String>.from(
+        document.data["stages"].map((x) => x["cropStage"].path));
```

Comment:
For readability, I would suggest using an actual name that represents the iterated item. Instead of x, please use 'each' or 'stage'.